src/main/java/org/apache/mesos/offer/OfferEvaluator.java

```diff
@@ -74,67 +56,73 @@ private boolean hasExpectedExecutorId(Offer offer, Protos.ExecutorID executorID)
         List<OfferRecommendation> creates = new ArrayList<>();
         List<OfferRecommendation> launches = new ArrayList<>();
 
-        ExecutorRequirement execReq = offerRequirement.getExecutorRequirement();
-        FulfilledRequirement fulfilledExecutorRequirement = null;
-        if (execReq != null) {
-            if (execReq.desiresResources() || execReq.getExecutorInfo().getExecutorId().getValue().isEmpty()) {
+        final Optional<ExecutorRequirement> execReq =
+                offerRequirement.getExecutorRequirementOptional();
+        Optional<FulfilledRequirement> fulfilledExecutorRequirement = Optional.empty();
+        Optional<ExecutorInfo> execInfo = Optional.empty();
+        if (execReq.isPresent()) {
+            execInfo = Optional.of(execReq.get().getExecutorInfo());
+
+            if (execReq.get().desiresResources()
+                    || execInfo.get().getExecutorId().getValue().isEmpty()) {
                 fulfilledExecutorRequirement = FulfilledRequirement.fulfillRequirement(
-                        execReq.getResourceRequirements(),
-                        execReq.getDynamicPortRequirements(),
+                        execReq.get().getResourceRequirements(),
+                        execReq.get().getDynamicPortRequirements(),
```

Comment:
`execReq.get()` appears often enough here that it might be worth extracting into its own variable.